### PR TITLE
Start assignment

### PR DESCRIPTION
**Pull request by ShaunWMusic**

### CODE REVIEW

**Comment by ShaunWMusic** [comment]
- [x] Create a new project named `20-ribbeter` using the `adonis new` command
- [x] Run `hub create` to create a new repository on Github
- [x] Run `yo adonis-broccoli` to add assets to the project
  - [x] Add the listed `scripts` for `assets:watch` and `assets:build`
- [x] Commit this as the "starting code"
- [x] Push `master` branch
- [x] Create `develop` branch
- [x] **Checkpoint:** Run `./ace auth:setup` to create User migrations and model!
- [x] **Checkpoint:** Run `./ace migration:run` to run the migrations to setup the database for users
- [x] **Checkpoint:** Create a `layout` view
  - [x] Add top nav
  - [x] Link a stylesheet to `/dist/app.css` to include SASS styles
  - [x] Make a block so that views that extend from `layout` can put content in to the page
  - [x] Style it!
- [x] **Checkpoint:** Create a `register` page
  - [x] Generate a User controller with `./ace make:controller`
  - [x] Setup router to respond to `/register` for `get` and `post` requests
  - [x] Setup view for `user.create` that extends from `layout`
  - [x] Make HTML for form using `form` helper (adding classes where needed)
  - [x] Style it!
- [x] **Checkpoint:** Store user when `register` is posted to
  - [x] Grab user input
  - [x] Try to:
    - [x] Store a new user with the form inputs
    - [x] Login the user automatically
    - [x] Send a success message
    - [x] Redirect to `/users`
  - [x] If there was an error, catch it
    - [x] Send back all inputs except `password`
    - [x] Send back an error message
    - [x] Redirect back
- [x] **Checkpoint:** Create a `login` page
  - [x] Generate a Login controller with `./ace make:controller`
  - [x] Setup router to respond to `/login` for `get` and `post` requests
  - [x] Setup view for `login.create` that extends from `layout`
  - [x] Make HTML for form using `form` helper (adding classes where needed)
- [x] **Checkpoint:** Store user when `login` is posted to
  - [x] Grab user input
  - [x] Try to:
    - [x] Attempt to login the user
    - [x] Send a success message
    - [x] Redirect to `/users`
  - [x] If there was an error, catch it
    - [x] Send back all inputs except `password`
    - [x] Send back an error message
    - [x] Redirect back
- [x] **Checkpoint:** Create a `users` page
  - [x] Setup router to respond to `/users` for `get` and `post` requests
  - [x] Setup view for `user.index` that extends from `layout`
  - [x] In controller get all users using the `User` model (`App/Model/User`)
  - [x] Send this user information to the view (note this will require calling `.toJSON()` on the users array)
  - [x] Using NJK loop through the users and show a card with the `username` and `email` (fake the picture with a picture of your favorite placholder image [square please])

**Comment by ShaunWMusic** [comment]
- [x] **Checkpoint:** Vue.js Setup
  - [x] Run `yo adonis-broccoli:vue` to add Vue to the project
  - [x] Add a `Route.all` render the `vue` template and protect this from non-logged-in users
- [x] **Checkpoint:** Add Adonis-Generators
  - [x] Use yarn to install `adonis-generators`
  - [x] Follow the documentation for installing `adonis-generators`
- [x] **Checkpoint:** Create Model, Migration, and Controller for Ribbit
  - [x] Using the Adonis Generators' Model Generator, create a `Ribbit` model and migration:
    - [x] Belongs to User
    - [x] String field for `body`
  - [x] Using Adonis Generators' Model Generator, create a `Ribbit` controller:
    - [x] Belongs to User
    - [x] String field for `body`
  - [x] Add a resource route so that requests to `/api/ribbits` use the `RibbitController`
- [x] **Checkpoint:** HTML Index Vue
  - [x] Using the `index.vue` file create HTML to represent the "Ribbit" page when visiting `/app` (the Vue router should already be working for this)
  - [x] Style this page
- [ ] **Checkpoint:** Submit New Ribbit
  - [ ] When a user submits the form on the left:
    - [ ] Make a POST request to `/api/ribbits` sending the contents of the form as `body`
    - [ ] Remember to add headers for `Content-Type`
    - [ ] Remember `credentials: 'same-origin'` to send cookies to prove that you are logged in
  - [ ] Update the `RibbitController`, it is expecting `user_id` to be part of the post data
    - [ ] Instead before creating the new Ribbit, set `input.user_id` to `request.currentUser.id` to grab the `id` of the logged in user
- [ ] **Checkpoint:** Show All Ribbits
  - [ ] When the vue application loads fetch all Ribbits from `/api/ribbits`
  - [ ] Loop through all "Ribbits" showing the username and body of the ribbit
- [ ] **Checkpoint:** Show New Ribbit Immediately
  - [ ] After a new ribbit has been saved, add the Ribbit to the array of all Ribbits

File: resources/styles/app.scss [hunk 1]
```diff
@@ -1,8 +1,211 @@
...
+@import 'scss/font-awesome';
 @import 'box-sizing';
 
-@import 'adonis';
+@import 'reset';
+
+.grid {
+  display: flex;
+  margin: -1rem;
+
```
Comment (rtablada) [inline resources/styles/app.scss:22]: Add `flex-wrap` on your grid

File: resources/styles/app.scss [hunk 2]
```diff
@@ -1,8 +1,211 @@
...
+  &--center {
+    justify-content: center;
+  }
+
+  &__item {
+    flex-basis: calc(15% - #{$m-gutter} - #{$m-gutter});
+    margin: $m-gutter;
+    box-shadow: 0.2rem 0.2rem 0.4rem rgba(#000, 0.4);
+
```
Comment (rtablada) [inline resources/styles/app.scss:32]: Add `flex-grow` to your grid__items

File: resources/javascript/routes/application.vue [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    };
+  },
+
+  mounted() {
+    this.findAllPosts();
+  },
+
+  methods: {
+    findAllPosts() {
+      fetch('/api/ribbit', {
```
Comment (rtablada) [inline resources/javascript/routes/application.vue:26]: This should fetch `/api/ribbits` right now I'm getting essentially a 404 error

**Review by rtablada — CHANGES_REQUESTED** [review]

**Pull request merged**